app/models/devise_token_auth/concerns/user.rb

```diff
@@ -147,7 +147,7 @@ def token_can_be_reused?(token, client)
       updated_at.to_time > Time.zone.now - DeviseTokenAuth.batch_request_buffer_throttle &&
 
       # ensure that the token is valid
-      DeviseTokenAuth::TokenFactory.valid_token_hash?(last_token)
+      DeviseTokenAuth::Concerns::User.tokens_match?(last_token, token)
```

Comment:
This should be in the factory, it doesn't make any sense to call a user's method for this

Reply:
Same method call is  used right in the previous method, and tokens_match? is defined only on the user.

Reply:
I think this doesn't make sense anymore as we have this in master https://github.com/lynndylanhurley/devise_token_auth/commit/4a103eddfa503354356704dcd2163e3cfc1a1f16